selectolax/utils.pxi

```diff
@@ -11,3 +17,91 @@ def preprocess_input(html, decode_errors='ignore'):
     if html_len > MAX_HTML_INPUT_SIZE:
         raise ValueError("The specified HTML input is too large to be processed (%d bytes)" % html_len)
     return bytes_html, html_len
+
+
+def do_create_tag(tag: str, parser_cls: ParserCls):
+    if not tag:
+        raise ValueError("Tag name cannot be empty")
+    return do_parse_fragment(f"<{tag}></{tag}>", parser_cls)[0]
+
+
+def get_fragment_type(
+    html: str,
+    parser_cls: ParserCls,
+    tree: Optional[Parser] = None,
+) -> Literal["document", "fragment", "head", "body", "head_and_body", "document_no_head", "document_no_body", "document_no_head_no_body"]:
+    if not tree:
+        tree = parser_cls(html)
+    
+    import re
+    html_re = re.compile(r"<html|<body|<head", re.IGNORECASE)
+
+    has_html = False
+    has_head = False
+    has_body = False
+    for match in html_re.finditer(html):
```

Comment:
To decide if HTML from user originally contained `<html>` and other tags, or whether those tags were added by Lexbor / Modest, I regex-search the original HTML.

This means that there could be false-positives if the given HTML contained e.g. word `<html>` inside a comment. But I wasn't able to find other approach..